### DISCUSSION TITLE
nestjs-logging

### DISCUSSION THREAD

**Discussion by giscus** [tomwray13/tomray.dev#53] [Blog Comments]
# nestjs-logging

Learn how to use the NestJS logger to add automatic HTTP request logging (and other cool capabilities) to your application. Also includes Pino setup.

https://www.tomray.dev/nestjs-logging

**Comment by peter-955** [comment]
Thank you for the detailed guide.
I'm facing an issue, that is when I receive a request from Postman, there will be 2 identical logs printed out. I'm not sure what is happening since I follow exactly as in the section`NestJS Logger with Pino`.
Could you please help me with this issue?

**Comment by barbatron** [comment]
Big thanks for the comprehensive and easy to follow guide! Got the logs for our new app sorted in minutes! Cheers! 🙏

**Comment by kafir-coder** [comment]
thanks for this article

I'm writing a system that logs all specific API calls and saves them into a document storage DB

**Comment by Maetes** [comment]
Thank you very much for this perfect guide. I have a sugestion for the section 

**Logs in production** and the related well as the related section

just change
```
        transport: {
          target: 'pino-pretty',
        },
```
to 
```
          transport: process.env.NODE_ENV !== 'production'
            ? { target: 'pino-pretty' }
            : undefined,
```
